packages/components/src/custom-select-control-v2/test/index.tsx

```diff
@@ -0,0 +1,176 @@
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+/**
+ * Internal dependencies
+ */
+import { CustomSelect, CustomSelectItem } from '..';
+
+describe( 'CustomSelect', () => {
+	it( 'Should be able to select multiple items when provided an array', async () => {
+		const user = userEvent.setup();
+
+		const defaultValues = [
+			'incandescent glow',
+			'ultraviolet morning light',
+		];
+
+		render(
+			<CustomSelect defaultValue={ defaultValues } label="Multi-select">
+				{ [
+					'aurora borealis green',
+					'flamingo pink sunrise',
+					'incandescent glow',
+					'rose blush',
+					'ultraviolet morning light',
+				].map( ( item ) => (
+					<CustomSelectItem key={ item } value={ item }>
+						{ item }
+					</CustomSelectItem>
+				) ) }
+			</CustomSelect>
+		);
+
+		const currentSelectedItem = screen.getByRole( 'combobox', {
+			expanded: false,
+		} );
+
+		expect( currentSelectedItem ).toHaveTextContent(
+			`${ defaultValues.length } items selected`
+		);
+
+		await user.click( currentSelectedItem );
+
+		expect( screen.getByRole( 'listbox' ) ).toHaveAttribute(
+			'aria-multiselectable'
+		);
+
+		defaultValues.map( ( value ) =>
+			expect(
+				screen.getByRole( 'option', { name: value, selected: true } )
+			).toBeVisible()
+		);
+
+		const nextSelection = screen.getByRole( 'option', {
+			name: 'rose blush',
+		} );
+
+		await user.click( nextSelection );
+
+		expect( nextSelection ).toHaveAttribute( 'aria-selected' );
+
+		expect( currentSelectedItem ).toHaveTextContent(
+			`${ defaultValues.length + 1 } items selected`
+		);
+	} );
+
+	it( 'Should be able to deselect items when provided an array', async () => {
+		const user = userEvent.setup();
+
+		const defaultValues = [
+			'aurora borealis green',
+			'incandescent glow',
+			'key lime green',
+			'rose blush',
+			'ultraviolet morning light',
+		];
+
+		render(
+			<CustomSelect defaultValue={ defaultValues } label="Multi-select">
+				{ defaultValues.map( ( item ) => (
+					<CustomSelectItem key={ item } value={ item }>
+						{ item }
+					</CustomSelectItem>
+				) ) }
+			</CustomSelect>
+		);
+
+		const currentSelectedItem = screen.getByRole( 'combobox', {
+			expanded: false,
+		} );
+
+		expect( currentSelectedItem ).toHaveTextContent(
+			`${ defaultValues.length } items selected`
+		);
+
+		await user.click( currentSelectedItem );
+
+		expect( screen.getByRole( 'listbox' ) ).toHaveAttribute(
+			'aria-multiselectable'
+		);
+
+		defaultValues.map( ( option ) =>
+			expect(
+				screen.getByRole( 'option', { name: option, selected: true } )
+			).toBeVisible()
+		);
```

Comment:
The assertions up to this point seem to be pretty much identical with what the previous test ("select multiple items when provided an array") verifies, so I'm not sure they're worth the maintenance cost? (No strong opinion)

```suggestion
		await user.click( currentSelectedItem );
```

Reply:
That's fair! I realize you are specifying the assertions, but just to note: we need to keep `await user.click( currentSelectedItem )` from this diff to ensure the popover is open for the deselection.  😄